flowmachine/flowmachine/features/location/majority_location.py

```diff
@@ -0,0 +1,56 @@
+from typing import List
+
+from flowmachine.core import Query
+from flowmachine.core.mixins import GeoDataMixin
+from flowmachine.features.utilities.subscriber_locations import BaseLocation
+
+
+class MajorityLocation(BaseLocation, Query):
+    def __init__(
+        self,
+        subscriber_location_weights: Query,
+        weight_column,
+        include_unlocatable=False,
+    ):
+        if weight_column not in subscriber_location_weights.column_names:
+            raise ValueError("weight_column must exist in subscriber_subset")
+        self.subscriber_location_weights = subscriber_location_weights
+        self.weight_column = weight_column
+        self.include_unlocatable = include_unlocatable
+        self.spatial_unit = subscriber_location_weights.spatial_unit
+        super().__init__()
+
+    @property
+    def column_names(self) -> List[str]:
+        # This should be a reduction, so no new columns
+        return [
+            "subscriber"
+        ] + self.subscriber_location_weights.spatial_unit.location_id_columns
```

Comment:
Could use `self.spatial_unit.location_id_columns` here (not that it really makes any difference)